sentry-core/src/main/java/io/sentry/core/MainEventProcessor.java

```diff
@@ -78,4 +79,8 @@ private void processNonCachedEvent(SentryEvent event) {
       event.setThreads(sentryThreadFactory.getCurrentThreads());
     }
   }
+
+  private boolean shouldApplyScopeData(final @Nullable Object hint) {
+    return (!(hint instanceof Cached) || (hint instanceof ApplyScopeData));
```

Comment:
Why isn't simply not `Cached` events get scoped applied? Do we have cases where it's cached but shouldn't apply. Is cached and should apply. Isn't cached but should apply and vice versa?

Reply:
this is how it was and this is actually the problem.
a captureMessage from NDK is a `Cached` hint, but still should apply scope's data, App is alive, so I composed `ApplyScopeData`

a captureException from NDK is the same case above.

now let's say there's a hard crash and App has died.
a captureException has a Cached hint and it should not apply scope's data, App has died and restarted.

Reply:
Got it. I just don't understand what's the use-case for Cached alone then.
To avoid caching the file again I guess?

Reply:
the answer above mentions that